lib/library.rb

```diff
@@ -0,0 +1,48 @@
+class Library
+    attr_reader :name,
+                :books,
+                :authors
+
+    def initialize(name, books = [] , authors = [])
+        @name = name
+        @books = books
+        @authors = authors
+    end
+    
+    def add_author(author)
+        @authors << author
+        @authors.map do |author|
+            author.books.each do |book|
+                @books << book
+            end
+        end
+        @authors
+    end
+
+    def publication_time_frame_for(author)
```

Comment:
Hopefully after the review you saw a less complicated way to do this. :) What you have here will work, but we can make it much more simple. This method though assumes that books are added in the order they were written, and that might not always be true.